### PR TITLE
Judge view of hearing schedule

### LINKED ISSUE #10409: Hearing Prep Merge | View schedule
Create judge view for viewing the schedule.

### PR DESCRIPTION
**Pull request by sharonwarner**
Resolves #10409 

### Description
This PR creates a judge view of hearing schedule. Judges are able to see hearing days if either the hearing day is assigned to them or the hearing day contains any hearings that are assigned to them.

### CODE REVIEW

File: app/models/hearing_day.rb [hunk 1]
```diff
@@ -123,6 +123,26 @@ def update_schedule(updated_hearings)
...
 
+    def upcoming_days_for_judge(start_date, end_date, user)
+      caseflow_assigned_hearing_days_and_hearings = HearingDay.includes(:hearings)
+        .where("DATE(scheduled_for) between ? and ?", start_date, end_date).select do |hearing_day|
+        hearing_day.judge == user || hearing_day.hearings.any? { |hearing| hearing.judge == user }
+      end
+
+      remaining_hearing_days = HearingDay.where("DATE(scheduled_for) between ? and ?", start_date, end_date)
+        .where.not(id: caseflow_assigned_hearing_days_and_hearings.pluck(:id)).order(:scheduled_for).limit(1000)
+
```
Comment (lomaxap) [inline app/models/hearing_day.rb]: do we need to exclude `caseflow_assigned_hearing_days_and_hearings`? is this so we have a unique list of days when you combine?

File: app/models/hearing_day.rb [hunk 2]
```diff
@@ -123,6 +123,26 @@ def update_schedule(updated_hearings)
...
+      vacols_hearings_for_remaining_hearing_days = HearingRepository.fetch_hearings_for_parents_assigned_to_judge(
+        remaining_hearing_days.pluck(:id), user
+      )
+
+      hearing_days_with_vacols_hearings = remaining_hearing_days.reject do |hearing_day|
+        vacols_hearings_for_remaining_hearing_days[hearing_day.id.to_s].nil?
+      end
+
+      caseflow_assigned_hearing_days_and_hearings + hearing_days_with_vacols_hearings
```
Comment (lomaxap) [inline app/models/hearing_day.rb]: this method was hard to follow for me. Is there are way to preload vacols hearings in the first query?
something like:
```ruby
hearing_days = HearingDay.where("DATE(scheduled_for) between ? and ?", start_date, end_date).includes(:hearings)
vacols_hearing_days = HearingRepository.fetch_hearings_for_parents(hearing_days.pluck(:id))
hearing_days_for_judge = hearing_days.select do |hearing_day|
   hearing_day.judge == user ||
   hearing_day.hearings.any? { |hearing| hearing.judge == user } ||
   vacols_hearing_days[hearing_day.id]&.any? { |hearing| some_user_id_comparison_for_vacols_hearing }
end
```

if something like that is possible, you could also get rid of the 2 new methods below

File: app/models/hearing_day.rb [hunk 3]
```diff
@@ -123,6 +123,26 @@ def update_schedule(updated_hearings)
       end
     end
 
+    def upcoming_days_for_judge(start_date, end_date, user)
+      caseflow_assigned_hearing_days_and_hearings = HearingDay.includes(:hearings)
+        .where("DATE(scheduled_for) between ? and ?", start_date, end_date).select do |hearing_day|
+        hearing_day.judge == user || hearing_day.hearings.any? { |hearing| hearing.judge == user }
```
[the remark this answers is not in the document]
Comment (sharonwarner) [reply app/models/hearing_day.rb]: Hearing.judge can't be nil if the HearingDay is not nil. The only time Hearing.judge is different from HearingDay.judge is if a hearing coordinator has overwritten the judge at the hearing level. They do not have the option to set the judge to nil.

File: app/models/hearing_day.rb (hunk 1 above)
Comment (sharonwarner) [reply app/models/hearing_day.rb]: I did this for performance. There's no need to pull hearings from VACOLS for hearing days we're already planning to display.

File: app/models/hearing_day.rb (hunk 2 above)
Comment (sharonwarner) [reply app/models/hearing_day.rb]: Yeah definitely agree this is cleaner. I guess it isn't significantly faster to pull ~30 fewer hearing days when we already have to pull a thousand or so. I'll update this.
